### ISSUE TITLE
4.0 - Crash on startup of project manager and startup of editor

### ISSUE THREAD

**Issue by voylin** [godotengine/godot#60766]
### Godot version

4.0.dev

### System information

Arch (Linux-lts 5.15.36-1) with i3 window manager

### Issue description

This issue probably only happens on tiling window managers I think.
Whenever I start up the project manager or a project, and right after that I switch my workspaces so the project/project manager startups in an inactive screen, Godot just doesn't open and crashes. This has been happening since Alpha 1.

This is the log I got just now:
![Screenshot from 2022-05-05 07-13-10](https://user-images.githubusercontent.com/56970759/166834739-633e4088-4d2c-48f5-ae79-ceb19aa6f848.png)
![Screenshot from 2022-05-05 07-13-43](https://user-images.githubusercontent.com/56970759/166834747-cebe9ee5-02e4-4412-aa98-ca277b3e6c87.png)

### Steps to reproduce

In a tiling window manager, open a project or the project manager and switch workspaces while the program is starting.

**Comment by Calinou** [comment]
Can you reproduce this if you start Godot with the `--single-window` command line argument?

**Comment by voylin** [comment]
> Can you reproduce this if you start Godot with the `--single-window` command line argument?

When using this, no crashes anymore. So should I just change my .desktop files to include this? I enabled the setting single window mode in the editor settings, but this is something different, is it?

**Comment by Calinou** [comment]
> So should I just change my .desktop files to include this? I enabled the setting single window mode in the editor settings, but this is something different, is it?

`--single-window` is exactly the same, although it can be used to enable single-window mode before you open the editor for the first time. (The editor setting is also used in the project manager.)

I recommend editing your `.desktop` file for now, but ideally, Godot should automatically use single-window mode if a tiling WM is detected. I don't expect a multi-window mode to actually work well when using a tiling WM.

**Comment by voylin** [comment]
> I recommend editing your `.desktop` file for now, but ideally, Godot should automatically use single-window mode if a tiling WM is detected. I don't expect Godot's multi-window mode to actually work well when using a tiling WM.

That's what I will do then for now. Should this issue stay open then until there is a patch for this?

**Comment by Calinou** [comment]
> Should this issue stay open then until there is a patch for this?

Yes, I'd prefer keeping this issue open until we manage to implement a fix.

**Comment by maniacalipsis** [comment]
Has the similar problem. All 4.0 alphas crashes when I switch to another desctop while the project manager startup or while the editor loads a project. The --single-window and enabling "Single Window" option in the editor settings has no effect.
Kubuntu 20.04
KDE Plasma 5.18.8
KDE Frameworks 5.68.0
QT 5.12.8
Kernel 5.4.0-113-generic

**Comment by MatthewZelriche** [comment]
This should be fixed by #64943

**Comment by voylin** [comment]
Can confirm that this is fixed now, no longer do I need to stop my multitasking behavior of flying through windows. Thank you!
